documentation/openapi-authoring-automated-guidelines.md

```diff
@@ -261,4 +263,302 @@ Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rul
+
+### <a name="R1007" />R1007 PatchInOperationName
+**Output Message**: 'PATCH' operation '{0}' should use method name 'Update'. Note: If you have already shipped an SDK on top of this spec, fixing this warning may introduce a breaking change.
+
+**Description**: Verifies whether value for `operationId` is named as per ARM guidelines.
+
+**Why the rule is important**: Per ARM guidelines, each 'PATCH' operation on a resource should have "update" in the name.
+
+**How to fix the violation**: Make sure that `operationId` is in the form of `NOUN_Update` or `Update`.
+
+**Impact on generated code**: Operation name in the generated SDK will be named based on this.
+
+**Examples**:
+* Resources_Update
+* Update
+
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="R3017" />R3017 GuidUsage
+**Output Message**: Guid used in model definition '{1}' for property '{0}'. Usage of Guid is not recommanded. If GUIDs are absolutely required in your service, please get sign off from the Azure API review board.
+
+**Description**: Verifies whether format is specified as "uuid" or not.
+
+**Why the rule is important**: Per ARM guidelines, GUID usage are discouraged.
+
+**How to fix the violation**: If GUIDs are absolutely required in your service, please get sign off from the Azure API review board.
+
+**Impact on generated code**: Based on each language generator, this may affect SDK.
+
+**Examples**: N/A
+
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="R1011" />R1011 HttpsSupportedScheme
+**Output Message**: Azure Resource Management only supports HTTPS scheme.
+
+**Description**: Verifies whether specification supports HTTPS scheme or not.
+
+**Why the rule is important**: All the ARM specification should support HTTPS endpoint.
+
+**How to fix the violation**: N/A.
+
+**Impact on generated code**: N/A.
+
+**Examples**: N/A
+
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="R2004" />R2004 NonApplicationJsonType
+**Output Message**: Please make sure that media types other than 'application/json' are supported by your service.
+
+**Description**: Verifies whether operation supports "application/json" as consumes or produces section.
```

Comment:
could you write a little more here, in terms of what happens if other formats are included, and why the warning is coming up, what should be done, and if there's any impact?